esp32/src/esp32_wifi.c

```diff
@@ -515,6 +521,7 @@ bool mgos_wifi_dev_sta_disconnect(void) {
     if (r == ESP_ERR_WIFI_NOT_INIT) r = ESP_OK; /* Nothing to stop. */
     if (r == ESP_OK) {
       s_started = false;
+      s_connecting = false;
```

Comment:
move it outside the condition, right after esp_wifi_disconnect - we should turn this flag off in ap+sta case as well. since we called esp_wifi_disconnect, we're definitely not connecting anymore.